src/lib/shim/src/lib.rs

```diff
@@ -44,18 +45,56 @@ pub fn simtime() -> Option<SimulationTime> {
     SimulationTime::from_c_simtime(unsafe { bindings::shim_sys_get_simtime_nanos() })
 }
 
-mod tls_allow_native_syscalls {
-    use super::*;
+/// Describes whether something is within the context of the shadow (shim) code,
+/// or the application/plugin code.
```

Comment:
nit: "Describes whether something is" - all of the methods on `ExecutionContext` rely on the global `CURRENT_EXECUTION_CONTEXT`, so is it possible to be more specific in this comment since it seems like `ExecutionContext` is only used for a single thing? For example something like "Controls the execution context of the shim, which may be within the context of the ..."?

Reply:
Done.